### PR TITLE
NIFIREG-234: Publish events for tenant CRUD

### PR DESCRIPTION
**Pull request by brosander**

### CODE REVIEW

**Comment by kevdoran** [comment]
Thanks for the contribution, @brosander. Will review...

**Comment by brosander** [comment]
Thanks @kevdoran it seems like there are sporadic failures around some of the tests, that's why all the close/reopen, my other pr had seemingly unrelated test failures as well...

**Comment by brosander** [comment]
Ex:

https://travis-ci.org/apache/nifi-registry/jobs/504770097

```
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated
```

File: nifi-registry-core/nifi-registry-provider-api/src/main/java/org/apache/nifi/registry/hook/EventFieldName.java [hunk 1]
```diff
@@ -26,6 +26,6 @@
     EXTENSION_BUNDLE_ID,
     VERSION,
     USER,
-    COMMENT;
-
+    USER_GROUP,
+    COMMENT
```
Comment (kevdoran) [inline nifi-registry-core/nifi-registry-provider-api/src/main/java/org/apache/nifi/registry/hook/EventFieldName.java]: I'n not sure we should reuse the EventFieldName.USER for the the user that was created. In other contexts it indicates the user that performed the operation (speaking of which, maybe we want that information for tenant CRUD events?)

I would suggest adding new EventFieldNames that align with the the tenant data model, such as `TENANT_ID` and `TENANT_IDENTITY` that can be used for all tenant types, or if we want to break it down into users and groups, `USER_ID`, `USER_IDENTITY`, `USER_GROUP_ID`, `USER_GROUP_IDENTITY`. What do you think?

File: nifi-registry-core/nifi-registry-framework/src/main/java/org/apache/nifi/registry/event/EventFactory.java [hunk 2]
```diff
@@ -133,4 +135,46 @@ public static Event extensionBundleVersionDeleted(final ExtensionBundleVersion b
                 .addField(EventFieldName.USER, NiFiUserUtils.getNiFiUserIdentity())
                 .build();
     }
+
+    public static Event userCreated(final User user) {
+        return new StandardEvent.Builder()
+                .eventType(EventType.CREATE_USER)
+                .addField(EventFieldName.USER, user.getIdentity())
```
[the remark this answers is not in the document]
Comment (brosander) [reply nifi-registry-core/nifi-registry-framework/src/main/java/org/apache/nifi/registry/event/EventFactory.java]: Good point, will add

File: nifi-registry-core/nifi-registry-provider-api/src/main/java/org/apache/nifi/registry/hook/EventFieldName.java (hunk 1 above)
Comment (brosander) [reply nifi-registry-core/nifi-registry-provider-api/src/main/java/org/apache/nifi/registry/hook/EventFieldName.java]: I think USER_ID, USER_IDENTITY, USER_GROUP_ID, USER_GROUP_IDENTITY more closely matches the terminology in object, methods of the tenant resource but don't have very strong feelings either way.

**Comment by brosander** [comment]
@kevdoran I think I've addressed the feedback, please let me know if I've missed anything or if further changes are needed

**Pull request closed without merging**